File: utils/mockAuth.ts
import { User } from '../types';

// Mock user data
const MOCK_USERS = [
  { uid: 'mock-user-1', phoneNumber: '+639123456789' },
  { uid: 'mock-user-2', phoneNumber: '+639876543210' },
];

// Mock OTP codes for testing
const MOCK_OTPS = {
  '+639123456789': '123456',
  '+639876543210': '654321',
};

class MockAuth {
  private currentUser: User | null = null;
  private confirmationResult: MockConfirmationResult | null = null;

  // Simulate auth state listener
  onAuthStateChanged(callback: (user: User | null) => void): () => void {
    // Return unsubscribe function
    return () => {};
  }

  // Mock sign in with phone number
  async signInWithPhoneNumber(phoneNumber: string): Promise<MockConfirmationResult> {
    // Validate phone number format
    if (!phoneNumber.startsWith('+63') || phoneNumber.length !== 13) {
      throw new Error('Invalid phone number format');
    }

    // Check if user exists in mock data
    const user = MOCK_USERS.find(u => u.phoneNumber === phoneNumber);
    if (!user) {
      throw new Error('Phone number not registered in mock data');
    }

    // Create mock confirmation result
    this.confirmationResult = new MockConfirmationResult(phoneNumber, user);
    return this.confirmationResult;
  }

  // Mock sign out
  async signOut(): Promise<void> {
    this.currentUser = null;
    this.confirmationResult = null;
  }

  // Get current user
  getCurrentUser(): User | null {
    return this.currentUser;
  }

  // Set current user (for testing)
  setCurrentUser(user: User | null): void {
    this.currentUser = user;
  }
}

class MockConfirmationResult {
  constructor(private phoneNumber: string, private user: User) {}

  async confirm(otp: string): Promise<any> {
    // Check if OTP is valid for this phone number
    const expectedOtp = MOCK_OTPS[this.phoneNumber as keyof typeof MOCK_OTPS];
    if (otp !== expectedOtp) {
      throw new Error('Invalid verification code');
    }

    // Simulate successful verification
    return {
      user: this.user,
    };
  }
}

// Export singleton instance
export const mockAuth = new MockAuth();

// Mock reCAPTCHA verifier (no-op)
export class MockRecaptchaVerifier {
  constructor(container: string, options?: any) {}

  render(): Promise<number> {
    return Promise.resolve(0);
  }

  clear(): void {}
}

// Export types for compatibility
export type { User };

File: click_ooh/lib/utils/mock_auth.dart
import '../types/types.dart';

// Mock user data
const List<Map<String, String>> mockUsers = [
  {'uid': 'mock-user-1', 'phoneNumber': '+639123456789'},
  {'uid': 'mock-user-2', 'phoneNumber': '+639876543210'},
];

// Mock OTP codes for testing
const Map<String, String> mockOtps = {
  '+639123456789': '123456',
  '+639876543210': '654321',
};

class MockConfirmationResult {
  final String phoneNumber;
  final User user;

  MockConfirmationResult(this.phoneNumber, this.user);

  Future<Map<String, dynamic>> confirm(String otp) async {
    // Check if OTP is valid for this phone number
    final expectedOtp = mockOtps[phoneNumber];
    if (otp != expectedOtp) {
      throw Exception('Invalid verification code');
    }

    // Simulate successful verification
    return {
      'user': user,
    };
  }
}

class MockAuth {
  User? _currentUser;
  MockConfirmationResult? _confirmationResult;

  // Simulate auth state listener
  void onAuthStateChanged(Function(User? user) callback) {
    // For simplicity, no real listener
  }

  // Mock sign in with phone number
  Future<MockConfirmationResult> signInWithPhoneNumber(String phoneNumber) async {
    // Validate phone number format
    if (!phoneNumber.startsWith('+63') || phoneNumber.length != 13) {
      throw Exception('Invalid phone number format');
    }

    // Check if user exists in mock data
    final userData = mockUsers.firstWhere(
      (u) => u['phoneNumber'] == phoneNumber,
      orElse: () => {},
    );

    if (userData.isEmpty) {
      throw Exception('Phone number not registered in mock data');
    }

    final user = User(
      uid: userData['uid']!,
      phoneNumber: userData['phoneNumber'],
    );

    // Create mock confirmation result
    _confirmationResult = MockConfirmationResult(phoneNumber, user);
    return _confirmationResult!;
  }

  // Mock sign out
  Future<void> signOut() async {
    _currentUser = null;
    _confirmationResult = null;
  }

  // Get current user
  User? getCurrentUser() {
    return _currentUser;
  }

  // Set current user (for testing)
  void setCurrentUser(User? user) {
    _currentUser = user;
  }
}

// Export singleton instance
final mockAuth = MockAuth();

// Mock reCAPTCHA verifier (no-op)
class MockRecaptchaVerifier {
  MockRecaptchaVerifier(String container, [Map<String, dynamic>? options]);

  Future<int> render() async {
    return 0;
  }

  void clear() {}
}